### ISSUE TITLE
[NTI] Interfaces with computed (symbol) properties incorrectly warning about no return statement

### ISSUE THREAD

**Issue by shicks** [google/closure-compiler#2530]
[Example][1]

```js
/** @interface */
class Foo {
  /** @return {Foo} */
  bar() {}

  /** @return {!Iterator<Foo>} */
  [Symbol.iterator]() {}
}
```

warns

```
input0:7: WARNING - Missing return statement. Function expected to return Iterator<Foo|null>.
  [Symbol.iterator]() {}
  ^^^^^^^^^^^^^^^^^^^^^^
```

It correctly understands `bar()` as being abstract, but it doesn't understand that `[Symbol.iterator]()` is abstract.

For now, the solution is to simply omit the computed property from the interface, since it's not correctly checked anyway, but we should eventually fix NTI to handle this correctly (it may come for free when we understand symbols).

[1]: https://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540interface%2520*%252F%250Aclass%2520Foo%2520%257B%250A%2520%2520%252F**%2520%2540return%2520%257BFoo%257D%2520*%252F%250A%2520%2520bar()%2520%257B%257D%250A%250A%2520%2520%252F**%2520%2540return%2520%257B!Iterator%253CFoo%253E%257D%2520*%252F%250A%2520%2520%255BSymbol.iterator%255D()%2520%257B%257D%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26MISSING_PROPERTIES%3D1%26TRANSPILE%3D1%26CHECK_TYPES%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26CLOSURE_PASS%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1
